Fix several typos in docs

MichaelHatherly opened the pull request:
- Fix incorrect indent in an "Examples" section.
- Remove some leading `.`s from two `.. function` signatures.
- Use `*`, not `_`, for emphasis in `eigs` docs.
- Replace `Note` section with an admonition in `eigs`.
- Add closing `|`s for absolute values in `gcdx` docs.

MichaelHatherly committed:
Fix several typos in docs

- Fix incorrect indent in an "Examples" section.
- Remove some leading `.`s from two `.. function` signatures.
- Use `*`, not `_`, for emphasis in `eigs` docs.
- Replace `Note` section with an admonition in `eigs`.
- Add closing `|`s for absolute values in `gcdx` docs.

tkelman commented on doc/stdlib/arrays.rst:
```diff
@@ -1045,7 +1051,7 @@ dense counterparts. The following functions are specific to sparse arrays.
 
 .. function:: permute{Tv,Ti,Tp<:Integer,Tq<:Integer}(A::SparseMatrixCSC{Tv,Ti}, p::AbstractVector{Tp},
-..............    q::AbstractVector{Tq})
+                  q::AbstractVector{Tq})
```
weren't these needed for genstdlib matching, or did you fix that?

MichaelHatherly replied:
The dots aren't needed and mess up the `HTML` rendering (see end of http://docs.julialang.org/en/latest/stdlib/arrays/), though `genstdlib` will match them with or without.

tkelman replied:
are #17005 https://github.com/JuliaLang/julia/pull/17036 mistaken then?

MichaelHatherly replied:
Misleading more than mistaken. "Wildcard" should probably be replaced with "single space", since, although `genstdlib` can deal with wildcards, Sphinx doesn't seem to like them and requires spaces.

tkelman replied:
could you add some clarifications / revision of that text then?

MichaelHatherly replied:
Will do.

MichaelHatherly replied:
Ok, I've updated the discussion in CONTRIBUTING.md to make it more obvious that spaces are needed. Hopefully `␣` characters displays correctly everywhere?

Sacha0 replied:
Thanks for catching this, and apologies for missing the Sphinx requirement! Best!

MichaelHatherly committed:
Update multiline signature matching explanation

Adds a more explicit example of the leading whitespace requirement for
docstring signatures in the RST docs.

tkelman commented on CONTRIBUTING.md:
````diff
@@ -104,21 +104,35 @@ If you want to edit an existing docstring signature, you **first** have to chang
+␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣qux, quux)
+```
+
+Note that the leading indent in the second line, `␣␣␣␣qux, quux)` in this example, must
````
leading indent relative to what starting point?

MichaelHatherly replied:
Updated.

tkelman commented on CONTRIBUTING.md:
````diff
@@ -104,21 +104,37 @@ If you want to edit an existing docstring signature, you **first** have to chang
 ```
-where `.` in the preceding snippet's second line is wildcard, and their number preceding indent-`    qux, quux)` matches the character count of `.. function:: `.
+..␣function::␣foo(bar, baz,
+␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣␣qux, quux)
````
I think I see why this was confusing me - firefox is displaying this character with an odd spacing, maybe a different character would be a better choice

MichaelHatherly replied:
That's unfortunate. How does `⋅` (`\cdot`) look:

```
..⋅function::⋅foo(bar, baz,
⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅⋅qux, quux)
```

?

Otherwise will probably just go back to `.` (fullstop), but note that it just represents a space, not a literal `.`.

tkelman replied:
Also not fixed-width, annoyingly.
![unicode](https://cloud.githubusercontent.com/assets/5934628/16714541/1ad8ed66-467b-11e6-93bf-5577d6ab02e5.png)

MichaelHatherly replied:
Oh well. Thanks for checking anyway.

MichaelHatherly replied:
Just going to use plain old spaces for this. Using `.` conflicts with the `.` in `.. function::` and would probably just cause more confusion.

MichaelHatherly commented:
Appveyor error looks unrelated. Good to merge?

MichaelHatherly commented:
Thanks.
